### PR TITLE
[AF-1792]  Migration of space information from system.git to config.git + HA improvements

### PR DESCRIPTION
**Pull request by adrielparedes**
# AF-1792

## Description

Several things happen in this PR that can't be split in several commits:

### Migration from system.git to config.git

All the information about spaces and repositories used to live in system.git project. That project makes that every change that was need to be done locks repository for all users. 
Now, all the information about each space is stored inside config.git in every space. So lock will happen only for that space and not for the whole file system. Also migration tool was modified to include previous changes.

### Logical Deletion

When you delete a Space or Repository, they are marked to be deleted. User will continue working as it was used to. Nothing from user perspectiva but they way that deleted spaces are shown. 
* A deleted Repository disappear from Library View and is marked to be deleted with a flag in config.git
* A deleted Space doesn't disappear from Spaces View but users can't get into that Space. In the UI the space is marked to be deleted. Also is marked to be deleted with a flag in config.git.

### Physical Deletion

Every one minute a FileSystemDeleteWorker runs to delete all the repositories and spaces that are marked to be deleted. This runs in all nodes, but only one will be able to execute those operations.

### HA Improvements

Many events where added to improve HA experience. Also visual improvements to show correct assets/spaces/repositories/contributors counters and deleted spaces right identification.

### RHSSO

RHSSO configuration was included to let BC consume contributors from SSO List.

## Collaborators

* Paulo Rego
* Pere Fernandez Pere
* Eder Ignatowicz
* Adriel Paredes

## PRs related

* https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/970
* https://github.com/kiegroup/appformer/pull/715
* https://github.com/kiegroup/kie-wb-common/pull/2692
* https://github.com/kiegroup/jbpm-wb/pull/1362
* https://github.com/kiegroup/kie-wb-distributions/pull/936

### CODE REVIEW

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

File: business-central-tests/business-central-tests-rest/src/test/java/org/kie/wb/test/rest/functional/SpacesScreenServiceIntegrationTest.java [hunk 1]
```diff
@@ -16,55 +16,81 @@
...
     @Before
     public void before() {
+        this.spaceName = SPACE_NAME + getRandomString();
         try {
-            client.deleteSpace(SPACE_NAME);
-        } catch (Exception ex){
+            Collection<Space> spaces = client.getSpaces();
+            spaces.forEach(space -> client.deleteSpace(space.getName()));
+        } catch (Exception ex) {
+            //ignore
```
Comment (tomasdavidorg) [inline business-central-tests/business-central-tests-rest/src/test/java/org/kie/wb/test/rest/functional/SpacesScreenServiceIntegrationTest.java]: I would at least log the exception.

File: business-central-tests/business-central-tests-rest/src/test/java/org/kie/wb/test/rest/functional/SpacesScreenServiceIntegrationTest.java [hunk 2]
```diff
@@ -16,55 +16,81 @@
...
+        }
+    }
+
+    @After
+    public void after() {
+        try {
+            Collection<Space> spaces = client.getSpaces();
+            spaces.forEach(space -> client.deleteSpace(space.getName()));
+        } catch (Exception ex) {
             //ignore
```
Comment (tomasdavidorg) [inline business-central-tests/business-central-tests-rest/src/test/java/org/kie/wb/test/rest/functional/SpacesScreenServiceIntegrationTest.java]: I would at least log the exception.

File: business-central-tests/business-central-tests-rest/src/test/java/org/kie/wb/test/rest/functional/SpacesScreenServiceIntegrationTest.java (hunk 1 above)
Comment (paulovmr) [reply business-central-tests/business-central-tests-rest/src/test/java/org/kie/wb/test/rest/functional/SpacesScreenServiceIntegrationTest.java]: +1

File: business-central-tests/business-central-tests-rest/src/test/java/org/kie/wb/test/rest/functional/SpacesScreenServiceIntegrationTest.java (hunk 2 above)
Comment (paulovmr) [reply business-central-tests/business-central-tests-rest/src/test/java/org/kie/wb/test/rest/functional/SpacesScreenServiceIntegrationTest.java]: +1

**Comment by adrielparedes** [comment]
@paulovmr @tomasdavidorg I've added some logs to that exception, I'm trying to see if I can reproduce those tests, but no luck so far. Do I have to do anything special to test to execute them than `mvn -Pbusiness-central-wildfly clean install`?

**Comment by paulovmr** [comment]
@adrielparedes I think it might be `mvn -Pbusiness-central,wildfly clean install` (comma instead of dash before wildfly).

**Comment by adrielparedes** [comment]
@paulovmr yes, sorry, that is what I meant to say. Anyway I can't reproduce them in my local, idk what I'm doing wrong

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

File: business-central-tests/business-central-tests-rest/src/test/java/org/kie/wb/test/rest/AccessRestTestBase.java [hunk 3]
```diff
@@ -40,7 +40,8 @@
     public AccessRestTestBase(User user) {
         this.user = user;
 
-        roleClient = RestWorkbenchClient.createWorkbenchClient(URL, user.getUserName(), user.getPassword());
+        int fiveMinutes = 5 * 60;
+        roleClient = RestWorkbenchClient.createWorkbenchClient(URL, user.getUserName(), user.getPassword(),fiveMinutes,fiveMinutes,fiveMinutes);
```
Comment (tomasdavidorg) [inline business-central-tests/business-central-tests-rest/src/test/java/org/kie/wb/test/rest/AccessRestTestBase.java]:
```suggestion
        roleClient = RestWorkbenchClient.createWorkbenchClient(URL, user.getUserName(), user.getPassword(), fiveMinutes, fiveMinutes, fiveMinutes);
```

**Comment by pefernan** [comment]
jenkins retest this please

**Pull request merged**